lib/clients/jira.go

```diff
@@ -337,24 +339,32 @@ func (j realJIRAClient) UpdateComment(issue jira.Issue, id string, comment githu
 // error. If it continues to fail until a maximum time is reached, it returns
 // a nil result as well as the returned HTTP response and a timeout error.
 func (j realJIRAClient) request(f func() (interface{}, *jira.Response, error)) (interface{}, *jira.Response, error) {
+	log := j.config.GetLogger()
+
 	var ret interface{}
 	var res *jira.Response
-	var err error
+	var reqErr error
 
 	op := func() error {
-		ret, res, err = f()
-		return err
+		ret, res, reqErr = f()
+		return reqErr
 	}
 
 	b := backoff.NewExponentialBackOff()
-	b.MaxElapsedTime = j.config.GetTimeout()
+	b.MaxElapsedTime = g.config.GetTimeout()
+
+	backoffErr := backoff.RetryNotify(op, b, func(err error, duration time.Duration) {
+		// Round to a whole number of milliseconds
+		duration /= retryBackoffRoundRatio // Convert nanoseconds to milliseconds
+		duration *= retryBackoffRoundRatio // Convert back so it appears correct
 
-	er := backoff.Retry(op, b)
-	if er != nil {
-		return ret, res, er
+		log.Errorf("Error performing operation; retrying in %v: %v", duration, err)
+	})
+	if backoffErr != nil {
+		return nil, nil, backoffErr
 	}
 
-	return ret, res, err
+	return ret, res, reqErr
```

Comment:
as stated above, do not use this error here. If you got this far, then reqErr must be nil.